apm/src/app/user/login/login.component.ts

```diff
@@ -0,0 +1,24 @@
+import { Component } from '@angular/core';
+import { Router } from '@angular/router';
+import { UserService } from '../user.service';
+
+@Component({
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.css']
+})
+
+export class LoginComponent{
+  userName: any;
+  password: any; 
+  homepageUrl: string = "/welcome";
+  constructor(private userService: UserService, private router: Router) { }
+
+  login(formValues: any) {
```

Comment:
Since we use typescript to help us avoid errors related to dynamic typing in javascript, we should avoid using any as a type. That does not help. Instead you can create an interface for the data you are expecting and use that

Reply:
Okay, I will create an interface.